pkg/controller/toolchainstatus/toolchainstatus_controller.go

```diff
@@ -400,6 +428,48 @@ func (r *ReconcileToolchainStatus) sendToolchainStatusUnreadyNotification(logger
 	return nil
 }
 
+func (r *ReconcileToolchainStatus) sendToolchainStatusRestoredNotification(logger logr.Logger,
```

Comment:
It looks like there's some opportunity for reuse between this function and `sendToolchainStatusUnreadyNotification`. It looks like only the name, subject and content are different so maybe we can create a new function eg. `sendToolchainStatusNotification` that accepts those as params and then both `sendToolchainStatusUnreadyNotification` and `sendToolchainStatusRestoredNotification` can use this function. The logic for validating the email address, removing the mangedfields, etc. can be in the common `sendToolchainStatusNotification` function.